lib/Basic/StringExtras.cpp

```diff
@@ -34,6 +34,17 @@ bool swift::canBeArgumentLabel(StringRef identifier) {
   return true;
 }
 
+StringRef swift::formattedArgumentLabel(StringRef identifier) {
+    bool newNameIsReserved = !canBeArgumentLabel(identifier);
+    llvm::SmallString<16> newStr;
+    if (newNameIsReserved)
+        newStr += "`";
+    newStr += identifier;
+    if (newNameIsReserved)
+        newStr += "`";
+    return newStr;
```

Comment:
This isn't safe; it's returning a StringRef to stack-allocated memory. It's better to populate a `SmallVectorImpl<char>` instead.

```c++
void swift::formattedArgumentLabel(SmallVectorImpl<char> &buffer, StringRef identifier)
```

Also, does it make more sense to put this as a free function, or on swift::Identifier?